src/effects/VST3/VST3Effect.cpp

```diff
@@ -350,58 +350,48 @@ bool VST3Effect::SupportsAutomation() const
 bool VST3Effect::SaveSettings(
    const EffectSettings &, CommandParameters & parms) const
 {
-   if(mEditController == nullptr)
-      return false;
-
-   using namespace Steinberg;
-   
-   for(int i = 0, count = mEditController->getParameterCount(); i < count; ++i)
-   {
-      Vst::ParameterInfo parameterInfo { };
-      if(mEditController->getParameterInfo(i, parameterInfo) == kResultOk)
+   return ForEachParameter
```

Comment:
Rewriting `SaveSettings` to use `ForEachParameter` should happen, as I said, in step 2.  Not this later step.

What is meant to change in this later step is the elimination of the use of the handles (here the editController).  Instead, EffectSettings is used, to find the already fetched values.

And because there is an assumption that values are fetched, this step shouldn't really be separated into three.  The insertion of all the necessary calls to `FetchSettings` should happen in the same commit -- otherwise this step in the transformation introduces an incorrectness, though fixed in a later step.